models/marts/core/product/fct_order_items.sql

```diff
@@ -0,0 +1,90 @@
+with orders as (
+    select * from {{ ref("stg_ecommerce__orders") }}
+),
+
+order_items as (
+    select * from {{ ref("stg_ecommerce__order_items") }}
+),
+
+order_reviews as (
+    select * from {{ ref("stg_ecommerce__order_reviews") }}
+),
+
+products as (
+    select * from {{ ref("stg_ecommerce__products") }}
+),
+
+customers as (
+    select * from {{ ref("stg_ecommerce__customers") }}
+),
+
+cancelled_missing_orders as (
+    select distinct order_id
+    from orders
+    where order_status not in ("canceled", "unavailable")
+),
+
+first_n_last_order as (
+    select
+        customers.customer_id,
+        min(orders.order_approved_at) as first_order_approved_at,
+        max(orders.order_approved_at) as last_order_approved_at,
+        date_diff(max(orders.order_approved_at), min(orders.order_approved_at), day) as customer_lifespan
+    from customers
```

Comment:
why do we need to start from the customer here? are we worried that some orders are not joinable to customers? Also this logic is very similar to that from the dimension table; if we needed to have one copy of that logic, would you put it here in the fact or in the other model in the dimension?

Reply:
@JFrackson I intended to create two versions of a fact table for purpose of practice. 
- dim_customers is more like a dimensionalized fact table as it contains a measure like customer lifespan, review score average etc.
- fct_order_items is more like a consolidated fact table
In a real-time project, It would certainly make sense to not have redundant logic.